File: test_vs/01_fibonacci.py
"""Benchmark: Recursive Fibonacci — Python baseline."""

import time


def fib(n: int) -> int:
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)


if __name__ == "__main__":
    start = time.perf_counter()
    result = fib(35)
    elapsed = time.perf_counter() - start
    print(f"fib(35) = {result}")
    print(f"Time: {elapsed:.4f}s")

File: benchmarks/cross_language/01_fibonacci.go
// Benchmark: Recursive Fibonacci -- Go

package main

import (
	"fmt"
	"runtime"
	"time"
)

func fib(n int) int {
	if n <= 1 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	runtime.GC()
	var memBefore runtime.MemStats
	runtime.ReadMemStats(&memBefore)

	start := time.Now()
	result := fib(35)
	elapsed := time.Since(start)

	var memAfter runtime.MemStats
	runtime.ReadMemStats(&memAfter)

	fmt.Printf("fib(35) = %d\n", result)
	fmt.Printf("__BENCH_METRICS__\n")
	fmt.Printf("wall_time_s=%.6f\n", elapsed.Seconds())
	fmt.Printf("peak_memory_kb=%.1f\n", float64(memAfter.TotalAlloc-memBefore.TotalAlloc)/1024.0)
}
